server/src/routes/api/v1/auth/index.ts

```diff
@@ -1,17 +1,24 @@
 import { Router, Request, Response, NextFunction } from "express";
 import { AuthService } from "./../../../../logic/services/auth.service";
 import { ErrorUtils } from "./../../../../logic/utils/errorUtils";
-import { ApiError, DbError, SocoboUser } from "./../../../../models/index";
+import { 
+  ApiError, DbError, SocoboUser, LoginResult 
+} from "./../../../../models/index";
 
 
 export class AuthRouteV1 {
   constructor (private _authService: AuthService, private _router: Router) {}
 
   createRoutes (): Router {
     // login the user
-    this._router.post("/login", (req: Request, res: Response, next: NextFunction) => {
-      this._authService.login(req.body.email, req.body.password)
-        .then((result: SocoboUser) => res.status(200).json(result))
+    this._router.post("/login", this.checkLoginRequest, (req: Request, res: Response, next: NextFunction) => {
+      // get data from request
+      let isEmailLogin: boolean = req.body.isEmailLogin;
```

Comment:
Hmm. At the first this is Backend code. :-)

In the middleware I check the request body and set the isEmailLogin flag. I don't like the fail check. I think this takes too much time.